Update the release lifecycle page to reflect current practices (#1200)
bitcoin-core/bitcoincore.org: _posts/en/pages/2016-01-15-lifecycle.md

Bitcoin Core releases are versioned as follows: MAJOR.MINOR, and release candidates are suffixed with rc1, rc2 etc.

We aim to make a major release every 6-7 months. These will be numbered 29.0, 30.0 etc.

> We will provide maintenance "minor releases" that fix bugs (security and otherwise) within the major releases. These

Reviewer: ```suggestion
We will provide maintenance releases, that fix bugs (security and otherwise), for each major release. These
```

will be numbered 29.3, 30.1, etc. We will not introduce major new features in maintenance releases (besides consensus
rules change, see below).

## Consensus rules

Proposals to change consensus rules are always shipped first in maintenance versions such as 22.2, 23.1 etc. This makes it easier for enterprise users to assess and test the proposal because of its smaller changeset compared to a major release. It also allows users who follow a more conservative upgrade path to adopt consensus rule changes in a more timely manner.

## Maintenance period

We always maintain the past three major versions released. Once a major version falls outside that window, it becomes
"End of Life". The threshold for backporting a change to a past major version increases as it ages. For example, if the
last major release is 30.0, then 29.x and 28.x are also considered maintained. Once 31.0 is released, then 28.x becomes
"End of Life".

> Major versions that are "End of Life" do not, as a general rule, receive security fixes. For more about our policy on

Reviewer 2: I don’t think it’s necessary to point out that we might make an exception and do more work than we generally intend to:

```suggestion
Major versions that are "End of Life" do not receive security fixes. For more about our policy on
```

Author: I think your alternative suggests that End of Life versions would never receive a security fix, no matter how critical. I think that's incorrect and my sentence better reflects our practices without setting too much expectations, so i didn't take your suggestion.

Reviewer 2: Perhaps:

```suggestion
Major versions that are "End of Life" generally do not receive security fixes. For more about our policy on
```

security fixes, see our [security advisories](/en/security-advisories/) page. We recommend running the latest

> maintenance release (point release) of the most recent major version you are able to upgrade to.

Reviewer 2: Previously, this is referred to as “minor release”. I would recommend using the same term to refer to the same concept:

```suggestion
maintenance release (minor release) of the most recent major version you are able to upgrade to.
```
